app/components/ThreadShareDialog.tsx

```diff
@@ -0,0 +1,205 @@
+import { useRef, useCallback, useState, useEffect } from "react";
+import {
+  Dialog,
+  useClipboard,
+  Portal,
+  Button,
+  CloseButton,
+  Input,
+  HStack,
+  Select,
+  createListCollection,
+  Field,
+  Link,
+} from "@chakra-ui/react";
+import { toaster } from "@/app/components/ui/toaster";
+import {
+  GlobeIcon,
+  LinkIcon,
+  LockIcon,
+  CopyIcon,
+  CheckIcon,
+} from "@phosphor-icons/react";
+import ChatDisclaimer from "./ChatDisclaimer";
+
+interface ThreadShareDialogProps {
+  threadId: string;
+  isPublic?: boolean;
+  onShare: (isPublic: boolean) => void;
+  isOpen: boolean;
+  onOpenChange: (open: boolean) => void;
+}
+
+type ValueChangeDetails = { value: string[] };
+
+function ThreadShareDialog(props: ThreadShareDialogProps) {
+  const { threadId, isPublic, onShare, isOpen, onOpenChange } = props;
+  const shareUrl =
+    typeof window !== "undefined"
+      ? `${window.location.origin}/app/threads/${threadId}`
+      : threadId;
```

Comment:
@kamicut I am not sure how to access the Vercel env vars here to correctly set the share URL prefix